test/e2e/framework/bind.go

```diff
@@ -54,25 +57,33 @@ func Bind(t *testing.T, authURLCh chan<- string, invocations chan<- SubCommandIn
 	require.NoError(t, err)
 	err = opts.Validate()
 	require.NoError(t, err)
-	opts.Runner = func(cmd *exec.Cmd) error {
-		bs, err := io.ReadAll(cmd.Stdin)
-		if err != nil {
-			return err
-		}
-		if invocations != nil {
-			invocations <- SubCommandInvocation{
-				Executable: cmd.Args[0],
-				Args:       cmd.Args[1:],
-				Stdin:      bs,
+
+	if !opts.DryRun {
+		opts.Runner = func(cmd *exec.Cmd) error {
+			bs, err := io.ReadAll(cmd.Stdin)
+			if err != nil {
+				return err
 			}
-		}
-		t.Logf("Running command: %s\nstdin:\n", cmd.String())
-		t.Logf("%s", bs)
+			if invocations != nil {
+				invocations <- SubCommandInvocation{
+					Executable: cmd.Args[0],
+					Args:       cmd.Args[1:],
+					Stdin:      bs,
+				}
+			}
+			t.Logf("Running command: %s\nstdin:\n", cmd.String())
+			t.Logf("%s", bs)
 
-		return nil
+			return nil
+		}
```

Comment:
why this special casing? This bind runner does not have to invoke a sub command. In that case the invocations channel will just not get any data.

Reply:
Runner is only executed if dry run is not enabled https://github.com/kube-bind/kube-bind/blob/a8214066c939e29ab648c94e4b74e62894a4fa62/pkg/kubectl/bind/plugin/bind.go#L314

in our case, test is failing when we pass `--dry-run`. Because `invocations` channel is not called by anybody(expectation is being called by runner)

Reply:
sorry, you are right. We just don't have to listen invocation channel and that is it.